### DISCUSSION TITLE
Adding the switch for rotating the vacuum layer direction of 2D material to x direction to accelerate the calculation

### DISCUSSION THREAD

**Discussion by xdzhu** [deepmodeling/abacus-develop#5274] [General]
### Background

For two-dimensional system in ABACUS, the parallel computation is faster along the _x_ direction. However, the vacuum layer is set in the _z_ direction in common two-dimensional material modeling.

### Describe the solution you'd like

Thus, I suggest that ABACUS can add a switch that automatically switches the direction of the vacuum layer to x direction internally. Before calculating the electronic structure, the structure is internally converted to the _x_ direction, and then ABACUS carry out the calculation. When all calculations are done, ABACUS transforms all output results to the original _z_ direction.

**Comment by mohanchen** [comment]
We will consider that.

**Comment by hongriTianqi** [comment]
- [ ] Review and understand the proposed feature and its importance.
- [ ] Research on the existing solutions and relevant research articles/resources.
- [ ] Discuss with the team to evaluate the feasibility of implementing the feature.
- [ ] Create a design document outlining the proposed solution and implementation details.
- [ ] Get feedback from the team on the design document.
- [ ] Develop the feature following the agreed design.
- [ ] Write unit tests and integration tests for the feature.
- [ ] Update the documentation to include the new feature.
- [ ] Perform code review and address any issues.
- [ ] Merge the feature into the main branch.
- [ ] Monitor for any issues or bugs reported by users after the feature is released.
- [ ] Address any issues or bugs reported by users and continuously improve the feature.
